examples/DivisorCurve.jl

```diff
@@ -0,0 +1,329 @@
+export AffineCurveDivisor, ProjCurveDivisor, iseffective, multiplicity,
+       divisor, curve, degree
+
+################################################################################
+
+abstract type CurveDivisor end
+
+################################################################################
+# The curve C is assume to be smooth and irreducible, but it is not checked.
+@doc Markdown.doc"""
+    AffineCurveDivisor(C::AffinePlaneCurve{S}, D::Dict{Point{S}, Int}) where S <: FieldElem
+
+Given a curve `C` which is assumed to be smooth and irreducible, return the divisor on the curve `C` defined by `D`.
+"""
+struct AffineCurveDivisor{S <: FieldElem} <: CurveDivisor
+    C::AffinePlaneCurve{S}
+    divisor::Dict{Point{S}, Int}
+    degree::Int
+    function AffineCurveDivisor(C::AffinePlaneCurve{S}, D::Dict{Point{S}, Int}) where S <: FieldElem
+        for P in keys(D)
+            ison_curve(P, C) || error("The point ", P.coord, " is not on the curve")
+            if iszero(D[P])
```

Comment:
Optional comment:

IMHO it is more elegant (and slightly more efficient) to just iterate over key-value pairs if you need both key and value; the same of course applies to the variant for the projective case.
```suggestion
        for (P, m) in D
            ison_curve(P, C) || error("The point ", P.coord, " is not on the curve")
            if m == 0
```